qiita_db/metadata_template/base_metadata_template.py

```diff
@@ -1157,3 +1157,23 @@ def update_category(self, category, samples_and_values):
                     % (category, value_str, value_types_str, column_type))
 
             raise e
+
+    def check_restrictions(self, restrictions):
+        """Checks if the template fulfills the restrictions
+
+        Parameters
+        ----------
+        restrictions : list of Restriction
+            The restrictions to test if the template fulfills
+
+        Returns
+        -------
+        set of str
+            The missing columns
+        """
+        def _col_iter():
+            for restriction in restrictions:
+                for col in restriction.columns:
+                    yield col
+
```

Comment:
I was just avoiding the double list comprehension, as sometimes it has been criticized. I think it is not too complicated, so I'll change it.